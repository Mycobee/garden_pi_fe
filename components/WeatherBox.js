import React from 'react';
import { Text, View, StyleSheet, Image, TouchableOpacity } from 'react-native';

const WeatherBox = ({ summary, highTemp, lowTemp, time, icon }) => (
  <View style={styles.widget}>
      <Text style={styles.bold}>{time}</Text>
    <View style={styles.icon}>
      {icon}
      <Text style={styles.summary}>{summary}</Text>
    </View>
    <View style={styles.statsContainer}>
      <Text style={styles.bold}>
        {Math.round(highTemp)}°F / {Math.round(lowTemp)}°F
      </Text>
    </View>
  </View>
)

const styles = StyleSheet.create({
  widget: {
    alignItems: 'center',
    alignItems: 'center',
    backgroundColor: '#d5fdd5',
    borderColor: '#A14550',
    borderRadius: 30,
    borderWidth: 2,
    justifyContent: 'space-between',
    margin: 3,
    padding: 10,
    width: 200
  },
  summary: {
    width: 90,
    marginLeft: 10,
  },
  icon: {
    width: 200,
    height: 80,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  bold: {
    fontSize: 15,
    fontWeight: 'bold'
  }
})

export { WeatherBox };