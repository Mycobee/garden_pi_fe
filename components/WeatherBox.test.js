import React from 'react';
import { WeatherBox } from './WeatherBox';
import 'react-native';
import renderer from 'react-test-renderer';


test('renders correctly', () => {
  const tree = renderer.create(<WeatherBox />).toJSON();
  expect(tree).toMatchSnapshot();
});