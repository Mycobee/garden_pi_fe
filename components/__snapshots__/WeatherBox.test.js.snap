// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d5fdd5",
      "borderColor": "#A14550",
      "borderRadius": 30,
      "borderWidth": 2,
      "justifyContent": "space-between",
      "margin": 3,
      "padding": 10,
      "width": 200,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "width": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "marginLeft": 10,
          "width": 90,
        }
      }
    />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      NaN
      °F / 
      NaN
      °F
    </Text>
  </View>
</View>
`;
